src/Core__Type.res

```diff
@@ -1,43 +1,109 @@
-type t = [#undefined | #object | #boolean | #number | #bigint | #string | #symbol | #function]
-
-external typeof: 'a => t = "#typeof"
-
-module Classify = {
-  type function
-  type object
-  type symbol
-
-  type t =
-    | Bool(bool)
-    | Null
-    | Undefined
-    | String(string)
-    | Number(float)
-    | Object(object)
-    | Function(function)
-    | Symbol(symbol)
-
-  @val external _internalClass: 'a => string = "Object.prototype.toString.call"
-  external _asBool: 'a => bool = "%identity"
-  external _asString: 'a => string = "%identity"
-  external _asFloat: 'a => float = "%identity"
-  external _asObject: 'a => object = "%identity"
-  external _asFunction: 'a => function = "%identity"
-  external _asSymbol: 'a => symbol = "%identity"
-
-  let classify = value => {
-    switch _internalClass(value) {
-    | "[object Boolean]" => Bool(_asBool(value))
-    | "[object Null]" => Null
-    | "[object Undefined]" => Undefined
-    | "[object String]" => String(_asString(value))
-    | "[object Number]" => Number(_asFloat(value))
-    | "[object Function]"
-    | "[object GeneratorFunction]"
-    | "[object AsyncFunction]" =>
-      Function(_asFunction(value))
-    | "[object Symbol]" => Symbol(_asSymbol(value))
-    | _ => Object(_asObject(value))
+type typeof = [
+  | #undefined
+  | #object
+  | #boolean
+  | #number
+  | #bigint
+  | #string
+  | #symbol
+  | #function
+]
+
+external typeof: 'a => typeof = "#typeof"
+
+type unknown
+external toUnknown: 'a => unknown = "%identity"
+
+type object // How to use with the Object module?
+type function
+
+// typeof null == "object". Gotta love that! Do better here.
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#typeof_null
+type jsType =
+  | Undefined
+  | Null
+  | Object(object)
+  | Bool(bool)
+  | Number(float)
+  | BigInt(Core__BigInt.t)
+  | String(string)
+  | Symbol(Core__Symbol.t)
+  | Function(function)
+
+external toObjectUnsafe: 'a => object = "%identity"
+external toBoolUnsafe: 'a => bool = "%identity"
+external toFloatUnsafe: 'a => float = "%identity"
+external toBigIntUnsafe: 'a => Core__BigInt.t = "%identity"
+external toStringUnsafe: 'a => string = "%identity"
+external toSymbolUnsafe: 'a => Core__Symbol.t = "%identity"
+external toFunctionUnsafe: 'a => function = "%identity"
```

Comment:
These are unnecessary as they can all be replaced by `Obj.magic`. The only difference is that `Obj.magic` is not bounded in its return value (which perhaps makes it the most unsafe function that exists), but that doesn't really matter since it's piped directly into a variant constructor that does exactly that.

Reply:
Obj.magic is not in Core. Haven't seen that one but love the name. If it is useful maybe it (with a scarier name) belongs in this module.

Reply:
It's in the underlying OCaml runtime. I don't think it should be exposed in core, it's not even properly documented for OCaml. It's not for end-users, but it's useful in well-maintained libraries such as this where the unsafety can be encapsulated.

See https://v2.ocaml.org/api/Obj.html

Reply:
Ok replacing with Obj.magic